lib/ui/painting.dart

```diff
@@ -861,9 +861,47 @@ class MaskFilter extends NativeFieldWrapperClass2 {
   ///
   /// A blur is an expensive operation and should therefore be used sparingly.
   MaskFilter.blur(BlurStyle style, double sigma) {
-    _constructor(style.index, sigma);
+    _constructorBlur(style.index, sigma);
   }
-  void _constructor(int style, double sigma) native "MaskFilter_constructor";
+  void _constructorBlur(int style, double sigma) native "MaskFilter_constructorBlur";
+
+  /// Creates a mask filter that implements a pair of shadows for an occluding
+  /// object - one representing ambient occlusion, and one representing a
+  /// displaced shadow from a point light.
+  ///
+  /// `occluderHeight`: Height of occluding object off of ground plane.
+  ///
+  /// `lightPos`: Position of the light applied to this object.
+  ///
+  /// `lightRadius`: Radius of the light (light is assumed to be spherical).
+  ///
+  /// `ambientAlpha`: Base opacity of the ambient occlusion shadow.
+  ///
+  /// `spotAlpha`: Base opacity of the displaced spot shadow.
+  MaskFilter.shadow({double occluderHeight: 0.0,
+                     double lightPosX: 0.0,
+                     double lightPosY: 0.0,
+                     double lightPosZ: 0.0,
+                     double lightRadius: 0.0,
+                     double ambientAlpha: 0.0,
+                     double spotAlpha: 0.0}) {
```

Comment:
nit: Our house style would be to have a newline after the `{` and then to indent the next line by two spaces more than the MaskFilter.shadow line.  We'd also a newline before the last `}` and indent the line with the `}` to match the MaskFilter.shadow line.